src/interceptors/ClientRequest/MockHttpSocket.ts

```diff
@@ -331,9 +353,22 @@ export class MockHttpSocket extends MockSocket {
     if (this.baseUrl.protocol === 'https:') {
       this.emit('secure')
       this.emit('secureConnect')
+
       // A single TLS connection is represented by two "session" events.
-      this.emit('session', Buffer.from('mock-session-renegotiate'))
+      this.emit(
+        'session',
+        this.connectionOptions.session ||
+          Buffer.from('mock-session-renegotiate')
+      )
       this.emit('session', Buffer.from('mock-session-resume'))
+
+      Reflect.set(this, 'encrypted', true)
+      // The server certificate is not the same as a CA
+      // passed to the TLS socket connection options.
+      Reflect.set(this, 'authorized', false)
```

Comment:
@mikicho, note that the `.authorized` must be set to `false`. It's false in a bypassed HTTPS communication as well. 

It will only be true if the certificate you pass to socket options is the same one that has signed the server you are requesting. That's false in most cases.